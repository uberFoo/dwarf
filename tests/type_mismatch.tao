fn main() {
    let a: int = 3.14;
    let b: float = 42;
    let c: int = "hello";
    let d: string = true;
    let e: bool = 42.0;
}
