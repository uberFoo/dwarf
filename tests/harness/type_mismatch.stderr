[31mError:[0m Type mismatch: expected `[3;35mint[0m`, found `[3;35mfloat[0m`.
   [38;5;246mâ•­[0m[38;5;246mâ”€[0m[38;5;246m[[0m type_mismatch:2:18 [38;5;246m][0m
   [38;5;246mâ”‚[0m
 [38;5;246m2 â”‚[0m [38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249ml[0m[38;5;249me[0m[38;5;249mt[0m[38;5;249m [0m[38;5;249ma[0m[38;5;249m:[0m[38;5;249m [0m[33mi[0m[33mn[0m[33mt[0m[33m [0m[33m=[0m[38;5;249m [0m[31m3[0m[31m.[0m[31m1[0m[31m4[0m[38;5;249m;[0m
 [38;5;240m  â”‚[0m            [33mâ”€[0m[33mâ”€[0m[33mâ”¬[0m[33mâ”€[0m[33mâ”€[0m [31mâ”€[0m[31mâ”€[0m[31mâ”¬[0m[31mâ”€[0m  
 [38;5;240m  â”‚[0m              [33mâ•°[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m expected [36m[3;35mint[0m[0m
 [38;5;240m  â”‚[0m                    [31mâ”‚[0m   
 [38;5;240m  â”‚[0m                    [31mâ•°[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m found [36m[3;35mfloat[0m[0m
[38;5;246mâ”€â”€â”€â•¯[0m

[31mError:[0m Type mismatch: expected `[3;35mfloat[0m`, found `[3;35mint[0m`.
   [38;5;246mâ•­[0m[38;5;246mâ”€[0m[38;5;246m[[0m type_mismatch:3:20 [38;5;246m][0m
   [38;5;246mâ”‚[0m
 [38;5;246m3 â”‚[0m [38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249ml[0m[38;5;249me[0m[38;5;249mt[0m[38;5;249m [0m[38;5;249mb[0m[38;5;249m:[0m[38;5;249m [0m[33mf[0m[33ml[0m[33mo[0m[33ma[0m[33mt[0m[33m [0m[33m=[0m[38;5;249m [0m[31m4[0m[31m2[0m[38;5;249m;[0m
 [38;5;240m  â”‚[0m            [33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”¬[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m [31mâ”€[0m[31mâ”¬[0m  
 [38;5;240m  â”‚[0m               [33mâ•°[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m expected [36m[3;35mfloat[0m[0m
 [38;5;240m  â”‚[0m                     [31mâ”‚[0m  
 [38;5;240m  â”‚[0m                     [31mâ•°[0m[31mâ”€[0m[31mâ”€[0m found [36m[3;35mint[0m[0m
[38;5;246mâ”€â”€â”€â•¯[0m

[31mError:[0m Type mismatch: expected `[3;35mint[0m`, found `[3;35mstring[0m`.
   [38;5;246mâ•­[0m[38;5;246mâ”€[0m[38;5;246m[[0m type_mismatch:4:18 [38;5;246m][0m
   [38;5;246mâ”‚[0m
 [38;5;246m4 â”‚[0m [38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249ml[0m[38;5;249me[0m[38;5;249mt[0m[38;5;249m [0m[38;5;249mc[0m[38;5;249m:[0m[38;5;249m [0m[33mi[0m[33mn[0m[33mt[0m[33m [0m[33m=[0m[38;5;249m [0m[31m"[0m[31mh[0m[31me[0m[31ml[0m[31ml[0m[31mo[0m[31m"[0m[38;5;249m;[0m
 [38;5;240m  â”‚[0m            [33mâ”€[0m[33mâ”€[0m[33mâ”¬[0m[33mâ”€[0m[33mâ”€[0m [31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”¬[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m  
 [38;5;240m  â”‚[0m              [33mâ•°[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m expected [36m[3;35mint[0m[0m
 [38;5;240m  â”‚[0m                     [31mâ”‚[0m     
 [38;5;240m  â”‚[0m                     [31mâ•°[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m found [36m[3;35mstring[0m[0m
[38;5;246mâ”€â”€â”€â•¯[0m

[31mError:[0m Type mismatch: expected `[3;35mstring[0m`, found `[3;35mbool[0m`.
   [38;5;246mâ•­[0m[38;5;246mâ”€[0m[38;5;246m[[0m type_mismatch:5:21 [38;5;246m][0m
   [38;5;246mâ”‚[0m
 [38;5;246m5 â”‚[0m [38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249ml[0m[38;5;249me[0m[38;5;249mt[0m[38;5;249m [0m[38;5;249md[0m[38;5;249m:[0m[38;5;249m [0m[33ms[0m[33mt[0m[33mr[0m[33mi[0m[33mn[0m[33mg[0m[33m [0m[33m=[0m[38;5;249m [0m[31mt[0m[31mr[0m[31mu[0m[31me[0m[38;5;249m;[0m
 [38;5;240m  â”‚[0m            [33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”¬[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m [31mâ”€[0m[31mâ”€[0m[31mâ”¬[0m[31mâ”€[0m  
 [38;5;240m  â”‚[0m                [33mâ•°[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m expected [36m[3;35mstring[0m[0m
 [38;5;240m  â”‚[0m                       [31mâ”‚[0m   
 [38;5;240m  â”‚[0m                       [31mâ•°[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m found [36m[3;35mbool[0m[0m
[38;5;246mâ”€â”€â”€â•¯[0m

[31mError:[0m Type mismatch: expected `[3;35mbool[0m`, found `[3;35mfloat[0m`.
   [38;5;246mâ•­[0m[38;5;246mâ”€[0m[38;5;246m[[0m type_mismatch:6:19 [38;5;246m][0m
   [38;5;246mâ”‚[0m
 [38;5;246m6 â”‚[0m [38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249ml[0m[38;5;249me[0m[38;5;249mt[0m[38;5;249m [0m[38;5;249me[0m[38;5;249m:[0m[38;5;249m [0m[33mb[0m[33mo[0m[33mo[0m[33ml[0m[33m [0m[33m=[0m[38;5;249m [0m[31m4[0m[31m2[0m[31m.[0m[31m0[0m[38;5;249m;[0m
 [38;5;240m  â”‚[0m            [33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”¬[0m[33mâ”€[0m[33mâ”€[0m [31mâ”€[0m[31mâ”€[0m[31mâ”¬[0m[31mâ”€[0m  
 [38;5;240m  â”‚[0m               [33mâ•°[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m expected [36m[3;35mbool[0m[0m
 [38;5;240m  â”‚[0m                     [31mâ”‚[0m   
 [38;5;240m  â”‚[0m                     [31mâ•°[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m found [36m[3;35mfloat[0m[0m
[38;5;246mâ”€â”€â”€â•¯[0m

